### PR TITLE
Dropping non context

### PR DESCRIPTION
**Pull request by Poolitzer**
This PR drops all non context based code. It also removes the regex handler (because it had non context code in it and *just* removing that one from the handler which was going to be removed anyway felt unnecessary).

### CODE REVIEW

**Comment by Poolitzer** [comment]
The tests will fail. I have no clue about mypy (help pls) and my internet connection is too unstable rn to run pytest, so I a going to use the CI results to fix the remaining tests

**Comment by Poolitzer** [comment]
Apart from the mypy errors mentioned above which I do not understand, I also dont understand this pytest error, all others pass \o/
```
__________________ TestDispatcher.test_one_context_per_update __________________

self = <tests.test_dispatcher.TestDispatcher object at 0x7f90e4361898>
dp = <telegram.ext.dispatcher.Dispatcher object at 0x7f90e50fcd90>

    def test_one_context_per_update(self, dp):
        def one(update, context):
            if update.message.text == 'test':
                context.my_flag = True
    
        def two(update, context):
            if update.message.text == 'test':
                if not hasattr(context, 'my_flag'):
                    pytest.fail()
            else:
                if hasattr(context, 'my_flag'):
                    pytest.fail()
    
        dp.add_handler(MessageHandler(Filters.regex('test'), one), group=1)
        dp.add_handler(MessageHandler(None, two), group=2)
        u = Update(1, Message(1, None, None, None, text='test'))
>       dp.process_update(u)

tests/test_dispatcher.py:142: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
telegram/ext/dispatcher.py:538: in process_update
    handler.handle_update(update, self, check, context)
telegram/ext/handler.py:147: in handle_update
    return self.callback(update, context)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

update = <telegram.update.Update object at 0x7f90e4248e88>
context = <telegram.ext.callbackcontext.CallbackContext object at 0x7f90e40d89a8>

    def two(update, context):
        if update.message.text == 'test':
            if not hasattr(context, 'my_flag'):
>               pytest.fail()
E               Failed

tests/test_dispatcher.py:134: Failed
```

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Overall looks good! A few notes:

* Searching for `use_context` in the branch still gives a number of results - please make sure to remove those
* I only skimmed the tests, but I'm sure they are fine :)
* Coverage decrease: Codecov marked nothing as untested that was tested before, IISC. I guess the decrease is just due to the fact that we're removing a lot of testing lines here so that the percentage of untested lines increases …

File: telegram/ext/__init__.py [hunk 1]
```diff
@@ -47,7 +47,6 @@
 from .filters import BaseFilter, MessageFilter, UpdateFilter, Filters
 from .messagehandler import MessageHandler
 from .commandhandler import CommandHandler, PrefixHandler
-from .regexhandler import RegexHandler
```
[the remark this answers is not in the document]
Comment (Poolitzer) [reply telegram/ext/__init__.py:50]: right.

**Review by harshil21 — CHANGES_REQUESTED** [review]
noticed that `regexhandler` needs to be removed from `telegram.ext.rst` too. Skimmed through the rest, seems fine.

_Poolitzer: Grrr, I swear I searched for regexhandler in the project_

**Comment by Poolitzer** [comment]
I tried fixing the merge conflicts but I might have broken everything :(

**Comment by Bibo-Joshi** [comment]
sorry to be bummer, but I think you were a little too thorough with your last doc change. IMO we should still state how the signature of the callback looks like, either by writing `def callback(…)` or just writing something like "the callback must be callable with two positional arguments of type Update and CallbackContext, respectively". What I meant was that we can skip the "for context based API" part, is there is no different API anymore now :)

**Comment by Poolitzer** [comment]
Hmm. Yeah I guess you are right, no way to know this without looking at the wiki.

**Pull request merged**